### PR TITLE
v3

### PR DESCRIPTION
**Pull request by PetrenkoAnton**

### CODE REVIEW

**Review by nau11713 — DISMISSED** [review]

**Review by nau11713 — DISMISSED** [review]

File: src/Http/HttpBaseClient.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+     */
+    protected function _send(BaseRequest $request): ResponseInterface
+    {
+        try {
+            return $this->httpClient->request($request->getMethod(), "." . $request->getEndpoint() .
+                $request->getParams(),
+                [
+                    "headers" => $request->getOptionsHeader($this->appToken),
+                    "body" => $request->getOptionsBody(),
+                    'debug' => $this->debug
```
Comment (SanjoDeundiak) [inline src/Http/HttpBaseClient.php]: What is that for?

File: src/Http/Request/Phe/EnrollRequest.php [hunk 2]
```diff
@@ -35,38 +35,45 @@
...
 
     /**
      * EnrollRequest constructor.
-     * @param string $endpoint
+     * @param AvailableRequest $request
      * @param int $version
-     * @param $appToken
      */
-    public function __construct(string $endpoint, int $version, $appToken)
+    public function __construct(AvailableRequest $request, int $version)
```
Comment (SanjoDeundiak) [inline src/Http/Request/Phe/EnrollRequest.php]: AvailableRequest should be moved either to the Client, or fixed in EnrollRequest (and others)

File: src/Pure/Storage/MariaDbPureStorage.php [hunk 3]
```diff
@@ -0,0 +1,1403 @@
...
+                $stmt->bindParam(2, $userId);
+                $userIdsArray[$j++] = $userId;
+
+                $stmt->execute();
+            }
+
+            if ($j != count($userIds))
+                throw new MariaDbOperationNotSupportedException();
+
+            // TODO! Add throwable
```
Comment (SanjoDeundiak) [inline src/Pure/Storage/MariaDbPureStorage.php]: Fix Todos

File: src/Pure/Storage/MariaDbPureStorage.php [hunk 4]
```diff
@@ -0,0 +1,1403 @@
...
+
+        if (empty($userIds))
+            return;
+
+        $userIdsArray = [];
+
+        try {
+            $conn = $this->getConnection();
+
+            $conn->beginTransaction();
```
Comment (SanjoDeundiak) [inline src/Pure/Storage/MariaDbPureStorage.php:979]: where is $conn->commit()?

File: src/Pure/Storage/MariaDbPureStorage.php [hunk 5]
```diff
@@ -0,0 +1,1403 @@
...
+
+        if (empty($roleNames))
+            return $roleCollection;
+
+        try {
+            $conn = $this->getConnection();
+
+            $namesSet = $roleNames;
+
+            // TODO: Proper StringBuilder size
```
Comment (SanjoDeundiak) [inline src/Pure/Storage/MariaDbPureStorage.php]: FIX

**Review by SanjoDeundiak — CHANGES_REQUESTED** [review]

File: tests/PureTest.php [hunk 6]
```diff
@@ -1195,7 +1195,7 @@ public function testRegistrationNewUserBackupsPwdHash(): void
     }
 
     /**
-     * @group
+     * @group f
```
Comment (SanjoDeundiak) [inline tests/PureTest.php]: ?

File: src/Http/HttpPureClient.php [hunk 7]
```diff
@@ -105,7 +104,8 @@ public function insertUser(InsertUserRequest $request): void
      */
     public function updateUser(UpdateUserRequest $request): void
     {
-        $this->_send($request);
+        $endpoint = sprintf("/update-user", $request->getUserId());
```
Comment (SanjoDeundiak) [inline src/Http/HttpPureClient.php]: This is very wrong

**Review by SanjoDeundiak — CHANGES_REQUESTED** [review]

**Review by SanjoDeundiak — APPROVED** [review]

**Pull request merged**